### ISSUE TITLE
Erratic build failures with "No space left on device"

### ISSUE THREAD

**Issue by j3mdamas** [travis-ci/travis-ci#8902]
Hi ~~guys~~,

We're having some erratic failures on Linux with the following:

```
/home/travis/.travis/job_stages: line 163: /tmp/tmp.UHjCqB8WRX/pids_after: No space left on device
awk: cmd. line:1: fatal: cannot open file `/tmp/tmp.UHjCqB8WRX/pids_after' for reading (No such file or directory)
```

Is anyone else facing this issue? Example: https://travis-ci.org/cuzzo87/htmd/jobs/315804789

**Comment by j3mdamas** [comment]
Updated example: https://travis-ci.org/Acellera/htmd/jobs/315934272

**Comment by BanzaiMan** [comment]
Please refrain from using "guys", as not all of us are.

As for your problem, do you have an estimate on how much disk space your build might be using? You can use the process described in https://docs.travis-ci.com/user/common-build-problems/#Troubleshooting-Locally-in-a-Docker-Image to get an estimate, if your local machine is not suitable.

**Comment by j3mdamas** [comment]
Hi @BanzaiMan! I apologize for the term, I was using it as to address [a group of people](https://en.wiktionary.org/wiki/guys). I will refrain from doing it in the future

Thanks a lot for the link, I will use that an test locally.

**Comment by j3mdamas** [comment]
Hi,
The docker image to test locally is very good, thanks! However, the points 8 and 9 are manual. Is there a way to reproduce the procedure that is run using the `.travis.yml` automatically?

**Comment by j3mdamas** [comment]
So, I finally ran the Docker image locally, by executing manually every step of our `.travil.yml`. The command at which the travis-ci build is failing here: https://travis-ci.org/Acellera/htmd/jobs/315934272 passes locally on my docker. After it, I do a `df -h` and I see there's still plenty of space left:

```
(travis-env) travis@26f1d459a75a:~/Acellera/htmd$ df -h
Filesystem           Size  Used Avail Use% Mounted on
overlay               50G   37G   14G  74% /
tmpfs                 64M     0   64M   0% /dev
tmpfs                 32G     0   32G   0% /sys/fs/cgroup
/dev/mapper/cl-root   50G   37G   14G  74% /etc/hosts
shm                   64M     0   64M   0% /dev/shm
tmpfs                 32G     0   32G   0% /sys/firmware
```
This is the command I ran locally: `docker run --name travis-debug -dit travisci/ci-garnet:packer-1490989530 /sbin/init`. I selected the docker image based on what the "worker information" showed me on the failing build.

Anything we're doing wrong?

**Comment by BanzaiMan** [comment]
How much disk space does your build demand?

**Comment by j3mdamas** [comment]
7.8 Gb

**Comment by BanzaiMan** [comment]
I suggest reducing disk space usage.

**Comment by j3mdamas** [comment]
What is the disk space limit that we have on travis-ci.org workers? Is that information available somewhere?

**Comment by j3mdamas** [comment]
Hi,

While trying to debug this issue, by attempting to reduce the amount of disk space used as recommended, I found another issue, which I reported on conda: https://github.com/conda/conda/issues/6481.

I am able to run that command locally, but not on the Travis workers, but let's see if the Conda team has some insight. Nonetheless, I saw other people reporting other IO issues with conda and Travis machines (https://github.com/conda/conda/issues/6473), so maybe this is a more generic symptom.

Thank you for your help
João
